pkg/apiserver/database/gorm/asset_scan.go

```diff
@@ -112,6 +112,13 @@ func (s *AssetScansTableHandler) CreateAssetScan(assetScan models.AssetScan) (mo
 		}
 	}
 
+	// Check the user wants to create a resource cleanup with other status than pending
```

Comment:
I think you need to check for nil values here:
`if assetScan.ResourceCleanup == nil || assetScan.ResourceCleanup.State == nil {...}`

Beside, if ResourceCleanup is not set by the user, we need to set it to Pending